lib/base-compiler.ts

```diff
@@ -179,8 +179,8 @@ export interface SimpleOutputFilenameCompiler {
     getOutputFilename(dirPath: string): string;
 }
 
-export class BaseCompiler implements ICompiler {
```

Comment:
This is the only class that implements ICompiler - which makes it worthless.
All other compilers inherit BaseCompiler itself.
Perhaps long ago the intention was different (?) but seems ICompiler will never really be used now.

Reply:
it was to get rid of circular references, please keep it

Reply:
@partouf please elaborate. Which circular references are present when it's removed?

Reply:
Apparently not currently, but before there were tools or other things that needed to compile something, and if it wanted to use the original compiler, you need something that's not actually `BaseCompiler`, but an interface that has the bare minimum to do so. I would like to keep the ability to do so in the future

Reply:
@partouf @jun   to implement any new compiler today you'd need waaaay more than the 4 functions described in `ICompiler`, and you probably won't need eg `cmake`. 

This interface seems a relic that would never be useful as is - if the need ever arises for an actual compiler interface we'd need to re-cut the minimum parts from BaseCompiler, but the current ICompiler is just dead code that we drag around and maintain. As part of the eternal battle against entropy I still think it's best we remove it.

Reply:
@mattgodbolt @jeremy-rifkin @AbrilRBS  WDYT?

Reply:
HI @OfekShilon ! Coming up for air here. I agree with @partouf 's thoughts on having a "more base" interface BUT if we're not using it currently, I'd be happy to remove and then reinstate something once we have a better idea what the interface needs to be. Happy to chat if you strongly disagree Partouf :)

Reply:
Agreed